kubernetes/GKE/README.md

```diff
@@ -0,0 +1,520 @@
+daemonset.apps/nvidia-driver-installer created
+```
+
+```bash
+kubectl get nodes "-o=custom-columns=NAME:.metadata.name,MEMORY:.status.allocatable.memory,CPU:.status.allocatable.cpu,GPU:.status.allocatable.nvidia\.com/gpu"
+``` 
+should show something similar to:
+
+```bash
+NAME                                        MEMORY       CPU     GPU
+gke-torchserve-default-pool-aa9f7d99-ggc9   12698376Ki   3920m   1
+```
+
+#### 2.3 Create a storage disk
+
+A standard storage class is created with google compute disk. If multiple pods need concurrent access to the same storage volume, you need Google NFS. Create the storage disk named *nfs-disk* with the following command:
+
+```bash
+gcloud compute disks create --size=200GB --zone=us-west1-a nfs-disk
+
+NAME     ZONE        SIZE_GB  TYPE         STATUS
+nfs-disk  us-west1-a  200      pd-standard  READY
+
+New disks are unformatted. You must format and mount a disk before it
+can be used. You can find instructions on how to do this at:
+
+https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
+
+```
+
+#### 2.4 Create NFS Server
+
+Modify the values.yaml in the nfs-provisioner with persistent volume name, disk name and node affinity zones and install nfs-provisioner using helm.
+
+```bash
+cd GKE
+
+helm install mynfs ./nfs-provisioner/
+```
+
+```kubectl get pods``` should show something similiar to:
+
+```bash
+NAME                                             READY   STATUS    RESTARTS   AGE
+pod/mynfs-nfs-provisioner-bcc7c96cc-5xr2k   1/1     Running   0          19h
+```
+
+#### 2.5 Create PV and PVC
+
+Run the below command and get NFS server IP:
+
+```bash
+kubectl get svc -n default mynfs-nfs-provisioner -o jsonpath='{.spec.clusterIP}'
+```
+
+Replace storage size and server IP in pv_pvc.yaml with the server IP got from above command. Run the below kubectl command and create PV and PVC
+
+```bash
+kubectl apply -f templates/pv_pvc.yaml -n default
+```
+
+Verify that the PVC / PV is created by excuting.
+
+```bash
+kubectl get pvc,pv -n default
+```
+
+Your output should look similar to
+
+```bash
+NAME                   CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM         STORAGECLASS   REASON   AGE
+persistentvolume/nfs   10Gi       RWX            Retain           Bound    default/nfs                           20h
+
+NAME                        STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+persistentvolumeclaim/nfs   Bound    nfs      10Gi       RWX                           20h
+```
+
+#### 2.6 Create a pod and copy MAR / config files
+
+Create a pod named `pod/model-store-pod` with PersistentVolume mounted so that we can copy the MAR / config files.
+
+```bash
+kubectl apply -f templates/pod.yaml
+```
+
+Your output should look similar to
+
+```bash
+pod/model-store-pod created
+```
+
+Verify that the pod is created by excuting.
+
+```bash
+kubectl get po
```

Comment:
@jagadeeshi2i The model-store pod is not getting created, using the commands specified Seeing following error in the describe pods: 

> (base) gchauhan-mbp:GKE gchauhan$ kubectl describe pod/model-store-pod
> Name:         model-store-pod
> Namespace:    default
> Priority:     0
> Node:         <none>
> Labels:       <none>
> Annotations:  kubectl.kubernetes.io/last-applied-configuration:
>                 {"apiVersion":"v1","kind":"Pod","metadata":{"annotations":{},"name":"model-store-pod","namespace":"default"},"spec":{"containers":[{"args"...
>               kubernetes.io/limit-ranger: LimitRanger plugin set: cpu request for container model-store
> Status:       Pending
> IP:           
> IPs:          <none>
> Containers:
>   model-store:
>     Image:      ubuntu
>     Port:       <none>
>     Host Port:  <none>
>     Command:
>       sleep
>     Args:
>       infinity
>     Requests:
>       cpu:        100m
>     Environment:  <none>
>     Mounts:
>       /pv from mypvc (rw)
>       /var/run/secrets/kubernetes.io/serviceaccount from default-token-8qsvr (ro)
> Conditions:
>   Type           Status
>   PodScheduled   False 
> Volumes:
>   mypvc:
>     Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
>     ClaimName:  nfs
>     ReadOnly:   false
>   default-token-8qsvr:
>     Type:        Secret (a volume populated by a Secret)
>     SecretName:  default-token-8qsvr
>     Optional:    false
> QoS Class:       Burstable
> Node-Selectors:  <none>
> Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
>                  node.kubernetes.io/unreachable:NoExecute for 300s
> Events:
>   Type     Reason             Age                  From                Message
>   ----     ------             ----                 ----                -------
>   Normal   NotTriggerScaleUp  2m57s                cluster-autoscaler  pod didn't trigger scale-up (it wouldn't fit if a new node is added):
>   Warning  FailedScheduling   14s (x3 over 2m59s)  default-scheduler   persistentvolumeclaim "nfs" not found
> 

Checking the pvc claim returns

> (base) gchauhan-mbp:GKE gchauhan$ kubectl get pvc,pv -n default
> NAME                                      STATUS   VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
> persistentvolumeclaim/model-store-claim   Bound    nfs      10Gi       RWX                           3m51s
> 
> NAME                   CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                       STORAGECLASS   REASON   AGE
> persistentvolume/nfs   10Gi       RWX            Retain           Bound    default/model-store-claim                           3m51s

Reply:
Had to change claim name from `nfs` to `model-store-claim` in the templates/pod.yaml file to make the above steps work